### ISSUE TITLE
Enhancement: Admin panel that allows easy user monitoring/UI configuration per user

### ISSUE THREAD

**Issue by devilsnare007** [danny-avila/LibreChat#3137]
### What features would you like to see added?

Allowing some sort of RBAC or ABAC control to manage users would be great while providing a per-user based UI customization.

### More details

This can not only enhance user monitoring, but also provide for a more easy to use UI per user according to the specific models/APIs allowed for said user. This should also allow customizing the UI for each user based on the models they're allowed to access.

### Which components are impacted by your request?

UI

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by muramasatheninja** [comment]
This would be extremely useful to me. I have some people I want to share my librechat instance with but don't want them to be able to burn through my credits with certain models. Being able to have permissions for models and apis would be great.

**Comment by VincentJGeisler** [comment]
agree, some form of user management panel is needed, to select which user can use what services etc, and approve new users before they can burn credits.

**Comment by GomoXU** [comment]
agree

**Comment by jasango** [comment]
Great work so far!  Admin Panel would be very useful, how can we know the expected release date?

**Comment by robertsmaoui** [comment]
Very intersting !!

**Comment by avdheshsolanki** [comment]
It is very much needed to have a clarity who is using your product.

**Comment by muhanstudio** [comment]
That would be a very good feature.

**Comment by mobyFrag** [comment]
Can't agree more! A user-friendly UI to list and manage users would be best 🤩

**Comment by vectorkovacspeter** [comment]
The only missing feature, for me

**Comment by jordisk** [comment]
@danny-avila do you have any ETA for that feature?

**Comment by harrisonhxy** [comment]
> This would be extremely useful to me. I have some people I want to share my librechat instance with but don't want them to be able to burn through my credits with certain models. Being able to have permissions for models and apis would be great.这对我来说非常有用。我想和一些人共享我的 librechat 实例，但又不想让他们通过某些模型消耗我的信用点数。如果能为模型和 apis 设置权限，那就太好了。

This is exactly the kind of functionality that I desperately need right now, as teachers and students within the subject group have different needs for endpoints, as well as different amounts of API keys assigned to them.So as an administrator, it is crucial to be able to limit the amount of endpoints a user can use, as well as set the API key amount for them!
By the way, this feature is not implemented yet right? Or I didn't find it.

**Comment by usnavy13** [comment]
This would be excellent and would pair perfectly with an api key subscription....  User management, statistics and the ability to allow select what endpoints, agents and assistants is sorely needed when working with a large number of users with different needs. 

Cant stress how much more useful this platform would be with these features.

**Comment by michnovka** [comment]
Hi @danny-avila , no pressure intended, but this was I think mentioned in the past on the 2024 EOY roadmap. I wonder what the updated roadmap is and when do you plan to get feature started? Thanks a lot for amazing work!

**Comment by cryptoguard** [comment]
Is there an update regarding this feature? I love everything about LibreChat; runs hosted on a server of mine, and would love to share it with other people but have them use their own API keys and not have access to my personal keys so they don't burn my credits.

**Comment by danny-avila** [comment]
Thanks for your patience. My goal is to get this done by the end of 2nd quarter this year, ideally well before then. Writing a new roadmap blog by the end of next week.

**Comment by AaronCreighton** [comment]
I have been wondering about the how to manage Bias/fairness.. possibly also generation of inappropriate content.. 

one possible method is to enable managers to see chat history.. E.g. HR can review chat history for HR. 

So, I am curious if some kind of team / Manager RBAC roles would be valuable.

**Comment by brettpappas** [comment]
Has work on this started yet? I would be interested in contributing to this feature around our needs to limit access to models and agent creation. Curious if the current planned changes would account for role based permissions on who can create agents as well as who they can share those agents with.

**Comment by tnsai** [comment]
If we're going down the road of roles and permissions, it would be great to add the ability to switch interface configurations depending on role (e.g. standard, power, admin users) or application (e.g. RAG vs. data capture vs. transactional agent, etc.) rather than having one globally applied configuration.

Also, adding unauthenticated anonymous user access alongside authenticated would open up many more use cases by providing different UIs for public versus private use.

**Comment by danny-avila** [comment]
>  it would be great to add the ability to switch interface configurations depending on role (e.g. standard, power, admin users) or application (e.g. RAG vs. data capture vs. transactional agent, etc.) rather than having one globally applied configuration.

This is planned.

> Also, adding unauthenticated anonymous user access alongside authenticated would open up many more use cases by providing different UIs for public versus private use.

Currently not planned but open to this.

**Comment by tnsai** [comment]
> > it would be great to add the ability to switch interface configurations depending on role (e.g. standard, power, admin users) or application (e.g. RAG vs. data capture vs. transactional agent, etc.) rather than having one globally applied configuration.
> 
> This is planned.

Fantastic.

> 
> > Also, adding unauthenticated anonymous user access alongside authenticated would open up many more use cases by providing different UIs for public versus private use.
> 
> Currently not planned but open to this.

I guess if SSO is being used from an external identity server then an external doorway service could be used to silently authenticate anonymous users under a single LC user account.  Not elegant but doable.

**Comment by benwallacestock** [comment]
Do we have a potential date that this might be released? We are currently looking at deploying this in our organisation and the admin ui might just be the thing that sells us on this route as the main product is fantastic!

**Comment by mikich-trintech** [comment]
That would be a valuable feature. While the admin panel is a suitable location for managing it, in my use case, it would be even more beneficial to limit or configure access to models and agents based on user properties, such as group or role membership, once SSO is enabled.

Thanks!

**Comment by paulkakell** [comment]
Any thoughts on an ETA for this?

**Comment by jasongilman** [comment]
Is there a way to manage access to users now through YAML or the database? I'd love to be able to do this for MCP server access.

**Comment by dobberr** [comment]
Any update on progress of the admin ui? Currently it's the only thing stopping me from switching to LibreChat

**Comment by Martin-SwDevOp** [comment]
> > Any update on progress of the admin ui? Currently it's the only thing stopping me from switching to LibreChat
> 
> I'm targeting March 31st (this month) to release this.

Wow, that would be amazing if it includes the possibility to assign models to certain user / user groups.

Ist that the case ? 🙂

**Comment by nagypite** [comment]
> > > Oh I see! I think the fact that there isn't an admin UI yet is why I hadn't yet discovered it's possible (because i have to sort through a whole config file, which isn't as visually intuitive)
> > 
> > 
> > You can do this easily with the UI via an agent. You can later pin an agent from the main model-selector dropdown
> 
> But does this apply globally to all users?

You can share the agent you created publicly with all users.

You can also set up modelSpecs in a way that only these agent will appear in the model selector.

**Comment by danny-avila** [comment]
Closing per the criteria @danny-avila set in the announcement above: "I will close this github issue when the admin panel is part of the default docker compose files, official docker images are published, and initial wave of community feedback."

All three are now met. `docker-compose.yml` on main ships an `admin-panel` service using the official image `registry.librechat.ai/clickhouse/librechat-admin-panel:latest`, and the panel has been public at [ClickHouse/librechat-admin-panel](https://github.com/ClickHouse/librechat-admin-panel) for about five months.

What that delivers against this thread: a GUI for everything previously in `librechat.yaml`, configuration profiles scoped to roles and groups, and group/role management with capability-based permissions. That covers the recurring asks here for per-role interface config, group support, and limiting which endpoints a given user can reach. The manual MongoDB queries several people shared as workarounds are no longer necessary.

Two parts of the original title are still in progress, and they now live in the panel repo rather than here: user management and user metrics. Feature requests and bug reports for the panel belong on [ClickHouse/librechat-admin-panel](https://github.com/ClickHouse/librechat-admin-panel), and general feedback in Discussion #11860.
